### ISSUE TITLE
Autoloading within custom installers doesn't work as expected if "target-dir" is used

### ISSUE THREAD

**Issue by df-jablan** [composer/composer#2079]
So here's the deal.  I've got this package with this [composer.json](https://github.com/dreamfactorysoftware/lib-platform-installer/blob/master/composer.json). As you can see, it requires "dreamfactory/lib-php-common-platform". This library has classes in the \DreamFactory\Platform\Utility namespace.

In a normal project, this is absolutely no problem thanks to Jordi's awesome Kung Fu. Multiple locations for a single namespace are completely supported via the "target-dir" option.

The problem here is that the above package also is targeted for the \DreamFactory\Platform\Utility namespace. But it seems the autoloader operates on the final output of composer, not the "work-in-progress" of what's been installed so far, as it were. 

So, in a nutshell, autoloading a class from a "required" package, from within a  custom installer fails -- even though the dependency has been installed.

I am not certain this is a bug or an undocumented feature.  Jordi, any guidance?
